client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useInterval } from 'lib/interval/use-interval';
+
+type TyperMode = 'TYPING' | 'DELETING';
+
+type AllTyperOptions = {
+	delayBetweenCharacters: number;
+	delayBetweenWords: number;
+};
+
+// if options are passed. At least one of them should be set
+type TyperOptions =
+	| Pick< AllTyperOptions, 'delayBetweenCharacters' >
+	| Pick< AllTyperOptions, 'delayBetweenWords' >
+	| AllTyperOptions;
+
```

Comment:
We don't get much value from this and it's pretty complicated. If someone really wants to pass `{}` for options that's not a problem. The types will say options isn't required anyways.

```suggestion
```